src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -36,9 +58,41 @@ internal sealed partial class NullableWalker : DataFlowPassBase<NullableWalker.L
 
         // PROTOTYPE(NullableReferenceTypes): Remove the Binder if possible. 
         private readonly Binder _binder;
-
         private readonly Conversions _conversions;
 
+        [Flags]
+        private enum MethodSignatureUse
+        {
+            None = 0,
+            ReturnType = 1,
+            ParameterTypes = 2,
+        }
+
+        /// <summary>
+        /// Indicates whether to use the return type and nullability from _methodSignatureOpt to
+        /// calculate return expression conversions, and whether to use the parameter types and
+        /// nullability from _methodSignatureOpt for initial parameter state. If those fields are not
+        /// used from _methodSignatureOpt, the signature of _member is used instead in each case.
+        /// </summary>
+        private readonly MethodSignatureUse _methodSignatureUse;
```

Comment:
Would it be simpler/clearer to have two bools: `_useReturnTypeFromMethodSignature` and `_useParameterTypesFromMethodSignature`?